### ISSUE TITLE
Model doesnt match the published results, ideas?

### ISSUE THREAD

**Issue by thh32** [euba/BacArena#120]
Using the standard model stated in the paper (Supplemental 1), the 'Strongly changing substances' showed very little variation over 10 hours or even 100 hours

[100 iterations.pdf](https://github.com/euba/BacArena/files/2318853/100.iterations.pdf)

data("Ec_core")
bac <- Bac(Ec_core)
arena <- Arena(n=20000, m=2000)
arena <- addOrg(arena,bac,amount=20)
arena <- addDefaultMed(arena, bac)
arena <- addSubs(arena, smax=0.5, mediac="EX_glc(e)", unit="mM")
eval <- simEnv(arena,time=20)
getVarSubs(eval)

getSubHist(eval, "EX_glc(e)")

par(mfrow=c(1,2))
plotCurves2(eval, legendpos = "topleft")

par(mfrow=c(2,5))
evalArena(eval, show_legend = FALSE, time=1:10)

When looking at the glc levels specifically there is a decrease from 2625000 to 2596943, however this can not be seen on the plot and does not match the large scale changes seen in the paper.

Is there a reason why the model doesnt match the paper in terms of substrate utilisation?

**Comment by jotech** [comment]
Hi @thh32 
thanks for pointing this out! Yes you are right this is not the old behavior. 

In short: The problem is that in your example the arena is too small and bacteria stop growing soon. That is why there is only a minor substrate uptake.

But it is possible to replicate the old results:
```R
library(BacArena)

data("Ec_core")
bac <- Bac(Ec_core, limit_growth=FALSE) # here is a change
arena <- Arena(n=20, m=20)
arena <- addOrg(arena,bac,amount=20)
arena <- addDefaultMed(arena, bac)
arena <- addSubs(arena, smax=0.5, mediac="EX_glc(e)", unit="mM", add=F) # another change
eval <- simEnv(arena,time=12)
```

Longer answer: In older versions we do not restrict growth even when cells could not duplicate anymore. This could lead to an accumulation of biomass. And since bigger cells take up more substrate this created _monster cells_ :) that showed an gigantic uptake. There was even an issue when the accumulated biomass got bigger than the maximum cell weight so that the growth model stoped working. (We fixed it by this commit: 6adc9e70330cac89f9b8a3af926ea87be4de494b)
To sum it up: The problem is the limited space and how you want to simulate the behavior in this case. Right now, cells are growing until their maximal biomass limit is reached and when they could not duplicate anymore they get nearly inactive because no further biomass gain is possible since their cell capacity is reached.
In case you have a setting in which this doesn't make sense, you can restore the old behavior by adding ``limit_growth=FALSE`` or you can also use a bigger value for maximal cell weight when creating and organism (``maxweight=1.172`` the unit is pico gram and the default value is for cells with similar size to E. coli).

Let me know if it worked for you and what you think about it.
And I will update the tutorial accordingly thank you for this!

**Comment by thh32** [comment]
Ahhh that makes sense and thanks for the detailed explaintion.

When running your code it seems to partially work as show below;
```
> getSubHist(eval, "EX_glc(e)")
         [,1]
1  125000.000
2  124798.764
3  124405.048
4  123467.154
5  120949.815
6  113149.928
7   87320.298
8   40920.660
9    3251.741
10      0.000
11      0.000
12      0.000
13      0.000
```

However as shown in the attached visualization of the 'Strongly changing substances' the huge utilization of glucose which is obvious from the `getSubHist` command, isnt visible. Any idea for the discrepancy?


[Example.pdf](https://github.com/euba/BacArena/files/2332516/Example.pdf)

**Comment by euba** [comment]
Hi,

Looking at the y axis scale it seems like there is a higher concentration in the new simulations. Try to add a different unit for the addition of metabolites in this line:
```
arena <- addDefaultMed(arena, bac, unit='fmol/cell')
```
Cheers, Eugen

**Comment by jotech** [comment]
I think the problem that it is not visible is due to the high concentration of the other substances added by ``addDefaultMed``. You can simply add a more defined medium (without using ``addDefaultMed``):
```R
arena <- addSubs(arena, smax=0.5, mediac="EX_glc(e)", unit="mM")
arena <- addSubs(arena, smax=1, mediac=c("EX_pi(e)", "EX_h2o(e)", "EX_o2(e)", "EX_nh4(e)"), unit="mM")
```
Or just print glucose alone
```R
plotCurves2(eval, subs = "EX_glc(e)")
```
